src/couch_index.erl

```diff
@@ -407,3 +394,160 @@ assert_signature_match(Mod, OldIdxState, NewIdxState) ->
         {Sig, Sig} -> ok;
         _ -> erlang:error(signature_mismatch)
     end.
+
+commit_compacted(NewIdxState, State) ->
+    #st{
+        mod=Mod,
+        idx_state=OldIdxState,
+        updater=Updater,
+        commit_delay=Delay
+    } = State,
+    {ok, NewIdxState1} = Mod:swap_compacted(OldIdxState, NewIdxState),
+    % Restart the indexer if it's running.
+    case couch_index_updater:is_running(Updater) of
+        true -> ok = couch_index_updater:restart(Updater, NewIdxState1);
+        false -> ok
+    end,
+    case State#st.committed of
+        true -> erlang:send_after(Delay, self(), commit);
+        false -> ok
+    end,
+    State#st{
+        idx_state=NewIdxState1,
+        committed=false
+     }.
+
+is_recompaction_enabled(IdxState, #st{mod = Mod}) ->
+    DbName = binary_to_list(Mod:get(db_name, IdxState)),
+    IdxName = binary_to_list(Mod:get(idx_name, IdxState)),
+    IdxKey = DbName ++ ":" ++ IdxName,
+    Global = config:get("view_compaction.recompaction", "enable"),
```

Comment:
I would prefer
```
[view_compaction]
enable_recompaction = true | false

[view_compaction.DbName]
enable = true | false
[view_compaction.DbName:IdxName]
enable = true | false
```

Because:
- This is boolean logic without else state;
- For others switches we use booleans as well;
- If we would like to add few more options for recompaction (still don't understand what is it and why), there wouldn't be any issues with;
- It's more error-prone solution (enable or enabled?)